### DISCUSSION TITLE
Questions about: Audio formats, recording/extraction lengths,  overlap, and data usage

### DISCUSSION THREAD

**Discussion by zackham02** [Nachtzuster/BirdNET-Pi#189] [Q&A]
These may have been asked and answered before but my search efforts have not resulted in me obtaining a clear understanding of certain options within BirdNet-Pi.

First, there are quite a few options for audio extraction format. I am currently using a RTSP stream as my mic input feed and I have heard/read that using wavPCM for the audio format is best for sound quality (as opposed to something like mp3) because it doesn't convert the audio  and subsequently compress the audio file as well.  Is this true? Does the audio extraction format you use affect detection ability or accuracy?  

Second, a little confused on extraction length. I have rec. length at 15ec and extract length at 14. Does this just mean that it will record for 14 seconds out of the 15 second detection time? Is 15/14 ok or too long/short? 

Third, overlap same thing. I am still not clear what overlap does despite researching a few different explanations about it online. I assume it must have to do with multiples bird calls happening at the same time or close by each other? Explain like I'm a complete jack a$$, because I am. 

Fifth, or fourth if you're paying attention. Data usage seems to be pretty, pretty, pretttttttttyyyyyyy (Larry David) high using this tool. Having had it running in my yard for 5 to 7  so far with 1900 total detections and 19 total species, my network usage shows  906.52 GiB received (close to a terabyte) so far already. That seems insanely large. I do have uncapped internet but am quite curious on this one. 

Thanks. I LOVE BirdNET and BirdNET-Pi !!!

**Comment by alexbelgium** [comment]
Hi, I've not been a user for super long but here is what I understood.

What is really important for the quality of detections is : 
- Good initial sound : 
-- Microphone with high Signal to Noise Ratio ; high sensitivity ; linear in the 20-15000 Hz range
-- Good location : avoid parasitic sounds such as leaves, wind, roads ...
- Good sound delivery
-- Avoid adding electrical noise through shielded cables, balanced audio, USB noise isolator...
-- Avoid lossy algorithms for your rtsp stream (usually not configurable): as the model is trained with pcm 16bits 48khz, this is what you should aim for. This is NOT something that can be configured in birdnet-pi : it can only be configured in the initial sound system - either the camera, or the rtsp stream server. Anyway, whatever sound you will provide (either higher end such as pcm24 bits or mp3) will be [converted](https://github.com/Nachtzuster/BirdNET-Pi/blob/714519b35bb5bd98a80aadcd016f06436bddb8cc/scripts/birdnet_recording.sh#L38) to pcm 16 bits 48khz. There is no link with the format setting from birdnet-pi that only defines how Birdnet-pi stores birdsongs after analysis
- Appropriate settings
-- Model version : seems like some models perform better in different regions, try what works best
-- Species range model : v1 seems more robust than v2.4
-- Species occurence threshold : use the Species List Tester to check the correct value for you)

As such for your answers : 
1. The format you select is how Birdnet-Pi stores the sound after analysis. Therefore, it has strictly no impact on the quality of detection. mp3 has a good ratio size vs quality.
2. To define extraction length, look at the size of the chunks analyzed in your logs. By default, Birdnet analyses 3 seconds chunks, so for example a 9 seconds extaction will be analyzed 0.0-3.0 ; 3.0-6.0 ; 6.0-9.0. But if you define an overlap of 0.5 seconds for example if will be : 0.0-3.0 ; 2.5-5.5 ; 5.0-8.0 ; 7.5-10.5 seconds. As you see, the last chunk goes beyond the size of your extracted sound, so the place between 9.0 and 10.5 will just be blank. This is not an issue, but a little waste of analysis. Also, having a short or long extraction length doesn't really matter on quality of detection - it just means you need to wait a little bit more but actually a longer length with a short overlap means that you don't lose a call in between two extraction sounds
3. Please see just above. The interest of overlap is to avoid a**hole birds who would decide to sing exactly in-between two chunks. For example, if you have 0 overlap and are analyzing 0.0-3.0 and 3.0-6.0 but the bird song is between 2.7 and 3.2s you would not detect it. Overlap of 0.5s allows to start the next chunk at 2.5s in this example, and catch this song
4. Birdnet-pi doesn't fetch any data online for its default feature that I'm aware of. What can use data : birdweather, and webui access.
-- Birdweather : as all your mp3 files will be uploaded to the birdweather site. My mp3 files are 0.3mo for 9 seconds. When I have 5000 detections per day, this mounts to 1.5Go
-- WebUI : a bit the same calculation except that it all depends when you access the site

My settings : https://github.com/alexbelgium/hassio-addons/blob/master/birdnet-pi/DOCS.md#app-settings-recommendation

Of course I'm more than happy to have my understandings challenged as we are all learning about this together!
Fully agree with you that Birdnet-Pi is really fantastic !
